alot/helper.py

```diff
@@ -23,20 +23,12 @@
 import magic
 
 
-def split_commandline(s, comments=False, posix=True):
+def split_commandline(s):
     """
-    splits semi-colon separated commandlines
+    splits semi-colon separated commandlines, ignoring quoted separators
     """
-    # shlex seems to remove unescaped quotes and backslashes
-    s = s.replace('\\', '\\\\')
-    s = s.replace('\'', '\\\'')
-    s = s.replace('\"', '\\\"')
-    lex = shlex.shlex(s, posix=posix)
-    lex.whitespace_split = True
-    lex.whitespace = ';'
-    if not comments:
-        lex.commenters = ''
-    return list(lex)
+    splitter = r'''((?:[^;"']|"(\\\\|\\"|[^"])*"|'(\\\\|\\'|[^'])*')+)'''
```

Comment:
>     splitter = r'''((?:[^;"']|"(\\\\|\\"|[^"])*"|'(\\\\|\\'|[^'])*')+)''' 
>                                                                     ^

Why do you match one or more groups? (`+`). This will not recognize the string `''`.

Reply:
The spec (from tests.test_helper.TestSplitCommandline.{test_simple,test_single}) does not want us to match empty strings.